content_scripts/doctolib/book.js

```diff
@@ -344,22 +405,32 @@
         if (slot === null) throw new Error("Aucun créneau disponible 3");
       }
 
-      // formats :
+      // formats france:
       // lun. 17 mai 08:54
       // ven. 13 août 09:10
       // jeu. 29 juil. 13:25
+
+      // formats allemagne:
+      // Do., 3. Feb., 08:25
+      slot.title = slot.title.replace(/,/giu, "");
+
       const parts = slot.title.match(
-        /([0-9]+) [\p{Letter}]+\.? ([0-9]+:[0-9]+)/gu
+        /([0-9]+)\.? ([\p{Letter}]+)\.? ([0-9]+:[0-9]+)/u
       );
       if (!parts) {
         throw new Error(
           `Impossible de cliquer sur le slot avec le titre ${slot.title}`
         );
       }
+
+      const currentMonth = new Date().getMonth();
+      const selectedMonth = monthToNumber(parts[2]);
+      const selectedDay = parseInt(parts[1]);
+      const selectedYear = estimateYear(currentMonth, selectedMonth);
+      const selectedTime = parts[3];
+
       const date = new Date(
```

Comment:
The date here is not really used for anything anymore, but this should fix it, there were little issues in the regex and the usage of `parts`. Additionally, the part with the year could have been incorrect when being at the end of a year, like now. The `estimateYear` function adds a year to the current one if the selected month is smaller than the current. This should work as long as we assume, that taking appointments in the past or in over a year should not be possible.

If I understand the code directly it was used before the 31st May 2020 to only search for available shots tomorrow. Should we remove the whole check?

Reply:
Sure! This rule doesn't exist in France anymore.

Reply:
I'll leave it in for now, because it seems useful for https://github.com/dunglas/vaccin.click/pull/86 :)